### PR TITLE
Add RobustAccuracy Metric

### PR DESCRIPTION
**Pull request by olliethomas**
I'm under no impression that this is good. Indexing by S in EM should be so much more elegant

### CODE REVIEW

**Review by MylesBartlett — APPROVED** [review]

**Comment by thomkeh** [comment]
Instead of
```
data.s[data.s.colums[0]]
```
you can do
```
data.s.iloc[:, 0]
```

File: ethicml/metrics/accuracy.py [hunk 1]
```diff
@@ -41,3 +41,25 @@ class F1(SklearnMetric):
...
+    @implements(SklearnMetric)
+    def score(self, prediction: Prediction, actual: DataTuple) -> float:
+        scores = []
+        for _s in actual.s[actual.s.columns[0]].unique():
+            preds_subset = Prediction(hard=prediction.hard[actual.s[actual.s.columns[0]] == _s])
+            data_subset = DataTuple(
+                x=actual.x[actual.s[actual.s.columns[0]] == _s],
+                s=actual.s[actual.s[actual.s.columns[0]] == _s],
+                y=actual.y[actual.s[actual.s.columns[0]] == _s],
+            )
```
Comment (thomkeh) [inline ethicml/metrics/accuracy.py:63]: Hm, I feel we must have a method for applying a mask to a data tuple

**Comment by olliethomas** [comment]
done - qq, do you know why the sklearnmetric is a tuple?

File: ethicml/metrics/accuracy.py [hunk 2]
```diff
@@ -41,3 +41,25 @@ class F1(SklearnMetric):
     sklearn_metric = (f1_score,)
     apply_per_sensitive: ClassVar[bool] = True
     _name: ClassVar[str] = "F1"
+
+
+@dataclass
+class RobustAccuracy(SklearnMetric):
+    """Classification accuracy."""
```
Comment (thomkeh) [inline ethicml/metrics/accuracy.py:48]:
```suggestion
    """Robust accuracy."""
```

File: ethicml/metrics/accuracy.py (hunk 1 above)
Comment (olliethomas) [reply ethicml/metrics/accuracy.py]: yeah, I'd have though so, we do it in enough places

**Comment by thomkeh** [comment]
Yes, it's explained here: https://github.com/predictive-analytics-lab/EthicML/blob/main/ethicml/metrics/accuracy.py#L20

**Comment by thomkeh** [comment]
But I'm not sure you gained much by subclassing SklearnMetric

**Comment by olliethomas** [comment]
So, technically this is a breaking change now as I've renamed `DataTuple.get_subset`, but i don't think anyone uses this directly anyway

**Comment by thomkeh** [comment]
So, the thing I was thinking of was `apply_to_joined` but that's probably overkill here. It would look like this:
```
subset = datatup.apply_to_joined(lambda df: df.loc[datatup.s == _s])
```

**Pull request merged**